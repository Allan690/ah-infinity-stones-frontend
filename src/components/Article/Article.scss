#article-comments input{
    width:100%;
}
#editArticleButton {
  background-color: #86C232;
  border-style: none;
  font-size: 1.5em;
}
.pull-right{
    padding-left: 5px!important;
}
.react-button {
    cursor: pointer!important;
    background-color:#838688!important;
    border: none;
}
.react-button i {
    color: white!important;
}
.react-button i:hover{
    color: #86C232!important;
}
.true {
    background-color:#86C232!important;
}
#msg{
    color: #86C232;
    display: block;
    margin-top: 25px;
    font-size: 1.2em;
}

.art-fun {
    padding-right: 30px;
    color: #86C232;

}
