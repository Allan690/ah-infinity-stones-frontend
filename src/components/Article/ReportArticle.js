/* eslint-disable rule camelcase */
/* eslint-disable camelcase */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ReportArticle extends Component {
  constructor(props) {
    super(props);
    this.handleReport = this.handleReport.bind(this);
    this.state = {
      reportMessage: '',
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleReport() {
    const { reportMessage } = this.state;
    const {
      props: {
        article: { art_slug },
        reportArticleAction,
      },
    } = this.props;
    const message = {
      artSlug: art_slug,
      report_msg: reportMessage,
    };
    reportArticleAction(message);
  }

  render() {
    const { reportMessage } = this.state;
    const modalContent = (
      <div className='modal-dialog modal-dialog-centered' role='document'>
        <div className='modal-content'>
          <div className='modal-header'>
            <h5 className='modal-title'>Report Article</h5>
            <button type='button' className='close' data-dismiss='modal' aria-label='Close'>
              <span aria-hidden='true'>&times;</span>
            </button>
          </div>
          <div className='modal-body'>
            <form id='reportMessageForm'>
              <input type='radio' name='reportMessage' value='Spam' onClick={this.onChange} />
              Spam
              <br />
              <input
                type='radio'
                name='reportMessage'
                value='Harrassment'
                onClick={this.onChange}
              />
              Harrassment
              <br />
              <input
                type='radio'
                name='reportMessage'
                value='Rules Violation'
                onClick={this.onChange}
              />
              Rules Violation
            </form>
          </div>
          <div className='modal-footer'>
            <button type='button' className='btn btn-secondary' data-dismiss='modal'>
              Cancel
            </button>
            <button
              type='button'
              className='btn btn-danger'
              id='modalSubmitReport'
              data-dismiss='modal'
              disabled={!reportMessage}
              onClick={this.handleReport}
            >
              Report
            </button>
          </div>
        </div>
      </div>
    );
    return (
      <div
        className='modal fade'
        id='reportArticleModal'
        tabIndex='-1'
        role='dialog'
        aria-labelledby='reportArticleModalTitle'
        aria-hidden='true'
      >
        {modalContent}
      </div>
    );
  }
}

ReportArticle.propTypes = {
  props: PropTypes.objectOf(PropTypes.objectOf(PropTypes.any)).isRequired,
};

export default ReportArticle;
