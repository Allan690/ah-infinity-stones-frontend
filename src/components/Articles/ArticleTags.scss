.body{
    display: block;
    display: -webkit-box;
    // max-width: 500px;
    height: 145px;
    font-size: 16px;
    line-height: 1.8;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;

}
.hght { height: 75vh !important; }
.art{
    float: right;
    padding: 1px 3px;
    margin-right: 15px;
    color:#86C232 !important;
    border: solid 1px #86C232 !important;
    &:hover {
        color: #fff !important;
    }
}

.read {
    margin-top: 20px;
}

.num {
    text-align: center;
    margin-bottom: 30px;
    margin-top: 20px;
    // margin: auto;
}
.art-link{
    &:visited {
        color:#86C232 !important;
    }
    color:#86C232;
}
