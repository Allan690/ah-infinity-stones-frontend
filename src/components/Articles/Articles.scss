@import url('https://fonts.googleapis.com/css?family=Hind+Siliguri');

.card {
    height: 170px;
}

.page-header {
    padding: 20px;
}

.article-preview {
    background: #F7FBF1 !important;
    font-family: 'Hind Siliguri', sans-serif;

    .card-header,
    .card-footer {
        background: transparent;
    }

    .card-header {
        border-bottom: none;
        padding: 10px;

        h5 {
            font-weight: 600;
            font-size: 18px;
            text-transform: capitalize;
            color: rgb(58, 57, 57);
        }
    }

    .card-body {
        padding: 4px 18px;
        max-height: 50px;
        overflow: scroll;

        &:first-letter {
            text-transform: uppercase;
        }
    }

    .card-footer {
        border-top: none;
    }

}
