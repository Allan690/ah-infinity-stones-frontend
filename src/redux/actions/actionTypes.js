export const REQUEST_LOGIN = 'REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'RECEIVE_LOGIN';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SOCIAL_LOGIN_REQUEST = 'SOCIAL_LOGIN_REQUEST';
export const SOCIAL_LOGIN_SUCCESS = 'SOCIAL_LOGIN_SUCCESS';
export const SOCIAL_LOGIN_ERROR = 'SOCIAL_LOGIN_ERROR';


export const REQUEST_INCREMENT = 'REQUEST_INCREMENT';
export const RECEIVE_INCREMENT = 'RECEIVE_INCREMENT';

export const PASSWORD_RESET_START = 'PASSWORD_RESET_START';
export const PASSWORD_RESET_COMPLETE = 'PASSWORD_RESET_COMPLETE';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';


export const PASSWORD_RESET_UPDATE_START = 'PASSWORD_RESET_UPDATE_START';
export const PASSWORD_RESET_UPDATE_COMPLETE = 'PASSWORD_RESET_UPDATE_COMPLETE';
export const PASSWORD_RESET_UPDATE_ERROR = 'ASSWORD_RESET_UPDATE_ERROR';
