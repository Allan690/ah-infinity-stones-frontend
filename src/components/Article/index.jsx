import React, {Component} from 'react'
import { connect } from 'react-redux';
import {requestIncrement}  from '../../redux/actions/articleActions';

class Articles extends Component {
    render(){
        return (
            <div>
              <h1>Article Counter</h1>
                <p>Count : {this.props.count} </p>
                <button onClick={this.props.requestIncrement}>Add Article</button>
            </div>
          )
    }
  
}
const mapStateToProps = state => ({ count: state.articleReducer.count });
const actionCreators = { requestIncrement };

export default connect(mapStateToProps, actionCreators)(Articles); 
