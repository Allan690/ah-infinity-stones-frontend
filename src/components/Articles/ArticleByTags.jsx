import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getByTagsRequest } from '../../redux/actions/articleActions';
import './ArticleTags.scss';


class ArticleByTags extends PureComponent {
  static propTypes = {
    match: PropTypes.instanceOf(Object).isRequired,
    results: PropTypes.instanceOf(Array).isRequired,
    getByTagsRequest: PropTypes.func.isRequired,
  };

  componentDidMount() {
    const {
      match: { params: { tag } },
      getByTagsRequest: getArticles,
    } = this.props;
    getArticles(tag);
  }

  render() {
    const { results, match: { params: { tag } } } = this.props;
    const noOfArticles = results.length;
    const elements = () => (
      results.map(article => (
        <div className="row hght" key={article.art_slug}>
          <div className="col-md-8">
            <h4 className="num">
              {`${noOfArticles} Articles Found With The Tag `}
              <strong>"{tag}"</strong>
            </h4>
            <h3>{article.title}</h3>

            <div className="row">
              <div className="col-sm-4">

                <h6>
                  <i className="mdi mdi-calendar-clock mdi-24px" />
                  &nbsp;
                  {article.read_time}
                  &nbsp;minute read
                </h6>
              </div>
              <div className="col-sm-4">
                {
                  article.tag.map(articleTag => (
                    <button type="button" className="btn btn-sm btn-outline-success art">
                      {articleTag}
                    </button>
                  ))
                }
              </div>

            </div>

            <i>{article.description}</i>
            <p className="body">{article.body}</p>
            <div className="read">
              <Link className="art-link" to={`/articles/${article.art_slug}`}>Read More...</Link>
            </div>
            <hr/>
          </div>
          <hr />
        </div>
      ))
    );

    return (
      <div className="row">
        <div className="col-md-4 num">
          Search Results
        </div>
        <div className="col-md-8">
          <div className="article">
            { elements() }
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { articleTagsReducer: { results, error } } = state;
  return { results, error };
};

const mapDispatchToProps = { getByTagsRequest };

export default connect(mapStateToProps, mapDispatchToProps)(ArticleByTags);
