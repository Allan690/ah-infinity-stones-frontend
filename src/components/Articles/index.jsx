import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getAllArticlesRequestAction } from '../../redux/actions/articleActions';


class GetAllArticles extends Component {
  componentDidMount() {
    this.props.getAllArticlesRequestAction();
  }

  render() {
    const articles = this.props.articles[0];
    const articlesList = articles && articles.length ? (
      articles.map((article) => {
        return (
          <div className="card bg-light text-dark" classstyle="width: 18rem;" key={article.art_slug}>
            <div className="card-header"><span className="card-title">{article.title}</span></div>
            <div className="card-body"><p className="card-text">{article.description}</p></div>
            <div className="card-footer text-center">
              <Link to={`/articles/${article.art_slug}`} className="btn btn-primary" id="read-more">Read More...</Link>
            </div>
          </div>
        );
      })
    ) : (<div className="card">No articles to show</div>);
    return (
      <div>
        <div className="page-header text-center"><h3>ALL ARTICLES</h3></div>
        <div className="container" id="allArticles"><div className="card-columns">{articlesList}</div></div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    articles: state.articleReducer.articles,
  };
};

const actionCreators = {
  getAllArticlesRequestAction,
};


export default connect(mapStateToProps, actionCreators)(GetAllArticles);
