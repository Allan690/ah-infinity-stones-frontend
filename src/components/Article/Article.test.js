/* eslint-disable no-unused-vars */
/* eslint-disable comma-dangle */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { call } from 'redux-saga/effects';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import store from '../../redux/store';
import Article from './index';
import like from '../../services/like';
import { sendLike } from '../../redux/actions/likeActions';
import { likeSaga } from '../../redux/sagas/likeSagas';
import likeReducer from '../../redux/reducers/likeReducer';
import reducer from '../../redux/reducers/articleReducer';
import * as types from '../../redux/actions/actionTypes';
import Articles from '../Articles/index';

configure({ adapter: new Adapter() });
describe('Single Article Component', () => {
  let articleContainer;
  let reduxArticle;
  const match = {
    params: {
      art_slug: 'js-testing',
    },
  };
  const article = {
    art_slug: 'smileys',
    title: 'smileys',
    rating_average: 4,
    description: 'emoji are working',
    body: '<p>thereâ€¦y are thats awespome</p>',
    read_time: 1,
  };
  beforeEach(() => {
    articleContainer = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Article match={match} article={article} />
        </BrowserRouter>
      </Provider>,
    );
  });
  it('Article renders successfully', () => {
    expect(articleContainer.exists()).toEqual(true);
  });
  it('render 1 Article component', () => {
    expect(articleContainer).toHaveLength(1);
  });
  it('article component should match snapshot', () => {
    expect(articleContainer.getElements()).toMatchSnapshot();
  });
});

describe('Article Reducer', () => {
  const payload = {
    art_slug: 'welcome-to-juja',
    title: 'The home of legends',
  };

  it('can handle CREATE_ARTICLE_SUCCESS', () => {
    expect(
      reducer(undefined, {
        type: types.CREATE_ARTICLE_SUCCESS,
        payload,
      }),
    ).toEqual({
      article: payload,
      articles: [],
      error: {},
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle CREATE_ARTICLE_FAILURE', () => {
    const payload = 'Could not create the article at this time';
    expect(
      reducer(undefined, {
        type: types.CREATE_ARTICLE_FAILURE,
        payload,
      }),
    ).toEqual({
      article: {},
      articles: [],
      error: payload,
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle GET_ALL_ARTICLES_SUCCESS', () => {
    const payload = 'Here are all the articles';
    expect(
      reducer(undefined, {
        type: types.GET_ALL_ARTICLES_SUCCESS,
        payload,
      }),
    ).toEqual({
      article: {},
      articles: payload,
      error: {},
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle GET_ALL_ARTICLES_FAILURE', () => {
    const payload = 'No articles found';
    expect(
      reducer(undefined, {
        type: types.GET_ALL_ARTICLES_FAILURE,
        payload,
      }),
    ).toEqual({
      article: {},
      articles: [],
      error: payload,
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle GET_ONE_ARTICLE_SUCCESS', () => {
    const payload = 'This is one article';
    expect(
      reducer(undefined, {
        type: types.GET_ONE_ARTICLE_SUCCESS,
        payload,
      }),
    ).toEqual({
      article: payload,
      articles: [],
      error: {},
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle GET_ONE_ARTICLE_FAILURE', () => {
    const payload = 'Could not fetch the article';
    expect(
      reducer(undefined, {
        type: types.GET_ONE_ARTICLE_FAILURE,
        payload,
      }),
    ).toEqual({
      article: {},
      articles: [],
      error: payload,
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle UPDATE_ARTICLE', () => {
    const payload = 'Could not update the article';
    expect(
      reducer(undefined, {
        type: types.UPDATE_ARTICLE,
        payload,
      }),
    ).toEqual({
      article: payload,
      articles: [],
      error: {},
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
  it('can handle DELETE_ARTICLE', () => {
    const payload = 'journey-to-the-moon';
    expect(
      reducer(undefined, {
        type: types.DELETE_ARTICLE,
        payload,
      }),
    ).toEqual({
      article: {},
      articles: [],
      error: {},
      header: 'ALL ARTICLES',
      rate: 0,
      rated: false,
      searchErrorMsg: '',
      specificArticle: {},
    });
  });
});
describe('Like functionality', () => {
  it('sends action of type SEND_LIKE', () => {
    const likeAction = sendLike({ like: 'False' }, 'sample-slug');
    expect(likeAction.type).toEqual('SEND_LIKE');
  });
  it('works with sagas', () => {
    const likeData = { likeData: { like: 'False' }, artSlug: 'sample-slug' };
    const generator = likeSaga(likeData);
    expect(generator.next().value).toEqual(call(like, likeData.likeData, likeData.artSlug));
  });
  it('has an initial state', () => {
    expect(likeReducer(undefined, { type: undefined })).toEqual({
      success: null,
      error: null,
      dislikeCount: null,
      likeCount: null,
      liked: null,
      disliked: null,
    });
  });
  it('it can send a like/dislike request', () => {
    expect(
      likeReducer(undefined, {
        type: types.SEND_LIKE,
      }),
    ).toEqual({
      success: null,
      error: null,
      dislikeCount: null,
      likeCount: null,
      liked: null,
      disliked: null,
    });
  });

  it('Like endpoint needs auth', async () => {
    const slug = 'like-this';
    const likeData = { likeData: { like: 'False' } };
    jest.setTimeout(50000);
    const data = await like(likeData, slug);
    expect(data.detail).toEqual('Your token is invalid. ');
  });

  it('can like an article successfully', () => {
    const resData = {
      Message: 'Thank you PromasterGuru12 for your opinion ',
      liked: true,
      likes_count: 1,
      disliked: false,
      dislikes_count: 0,
    };
    expect(
      likeReducer(undefined, {
        type: types.LIKE_SUCCESS,
        data: resData,
      }),
    ).toEqual({
      success: 'Thank you PromasterGuru12 for your opinion ',
      error: null,
      dislikeCount: 0,
      likeCount: 1,
      liked: true,
      disliked: false,
    });
  });
  it('can handle like error', () => {
    const error = 'Could not like this article';
    expect(
      likeReducer(undefined, {
        type: types.LIKE_ERROR,
        error,
      }),
    ).toEqual({
      success: null,
      error: 'Could not like this article',
      dislikeCount: null,
      likeCount: null,
      liked: null,
      disliked: null,
    });
  });
});
