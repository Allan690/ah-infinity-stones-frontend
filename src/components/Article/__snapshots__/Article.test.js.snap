// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Article Component article component should match snapshot 1`] = `
Array [
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Connect(Article)
        article={
          Object {
            "art_slug": "smileys",
            "body": "<p>thereâ€¦y are thats awespome</p>",
            "description": "emoji are working",
            "rating_average": 4,
            "read_time": 1,
            "title": "smileys",
          }
        }
        match={
          Object {
            "params": Object {
              "art_slug": "js-testing",
            },
          }
        }
      />
    </BrowserRouter>
  </Provider>,
]
`;
